launcher/InstanceList.cpp

```diff
@@ -947,9 +945,12 @@ QString InstanceList::getStagedInstancePath()
 
 bool InstanceList::commitStagedInstance(const QString& path,
                                         InstanceName const& instanceName,
-                                        const QString& groupName,
+                                        QString groupName,
                                         InstanceTask const& commiting)
 {
+    if (groupName.isEmpty() && !groupName.isNull())
+        groupName = QString();
```

Comment:
does this line do something?
As you already verified that the group name is empty and not null. So why recreate it?

Reply:
this checks whether it's "" and changes it to null